libs/ngxtension/map-skip-undefined/src/map-skip-undefined.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { from, of, toArray } from 'rxjs';
+import { filterUndefined, mapSkipUndefined } from './map-skip-undefined';
+
+describe(filterUndefined.name, () => {
+	it('given an observable of null, undefined, 42, filter out the undefined, emit null and 42', (done) => {
+		const in$ = of(null, undefined, 42);
+		const out$ = in$.pipe(filterUndefined());
+
+		out$.pipe(toArray()).subscribe((r) => {
+			expect(r).toEqual([null, 42]);
+			done();
+		});
+	});
+});
+
+describe(mapSkipUndefined.name, () => {
+	it('given an observable >1-42-3| and a mapping function that double ONLY the odds value, then result is an observable of >"2"--"6"-| the intial even value (42) is not mapped and so filtered out', (done) => {
+		const in$ = from([1, 42, 3]);
+		const out$ = in$.pipe(
+			mapSkipUndefined((n) => {
+				if (n % 2) return String(n * 2);
+				else return undefined; // explict return undefined to skipout (filter) some value from the out observable
+			})
+		);
+
+		out$.pipe(toArray()).subscribe((r) => {
+			expect(r).toEqual(['2', '6']);
+			done();
+		});
+	});
```

Comment:
simple questio: why did you go with 'toArray' instead of observer-spy?

Reply:
@tomalaforge observer-spy commit was reverted due to some ESM issue.

Reply:
observer-spy commit was reverted due to some ESM issue

Reply:
@tomalaforge yes I see you commit to introduce observer-spy, and later @nartc revert it
So I was confused, and decided to go to _"easy way"_ rewriting test with `toArray` as you suggested 👍